### PR TITLE
Nomis api

### PR DESCRIPTION
**Pull request by dan-palmer**

### CODE REVIEW

File: lib/nomis/client.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    def get(path:, prison_number:)
+      connection.get(path, noms_id: prison_number).body
+    rescue Faraday::Error::TimeoutError => error
+      raise Nomis::Error::RequestTimeout, error
+    rescue JSON::ParserError => error
+      raise Nomis::Error::InvalidResponse, error
+    end
+
+    ENDPOINT = 'https://serene-chamber-74280.herokuapp.com/'
```
Comment (whoojemaflip) [inline lib/nomis/client.rb]: could this come from the environment config?

File: spec/lib/nomis/client_spec.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+require 'nomis/error'
+
+WebMock.disable_net_connect!(allow: 'codeclimate.com')
+
+RSpec.describe Nomis::Client do
+  subject { described_class.new }
+
+  describe '#offender_details' do
+    before do
+      stub_request(:get, 'https://serene-chamber-74280.herokuapp.com/offender_details?noms_id=A1401AE').
```
Comment (whoojemaflip) [inline spec/lib/nomis/client_spec.rb]: this magic string again

File: spec/lib/nomis/client_spec.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        it 'raises a RequestTimeout error' do
+          expect { subject.offender_details(prison_number: 'A1401AE') }.
+            to raise_error(Nomis::Error::RequestTimeout)
+        end
+      end
+
+      context 'invalid response' do
+        before do
+          stub_request(:get,'https://serene-chamber-74280.herokuapp.com/offender_details?noms_id=A1401AE').
+            to_return(body: 'Invalid')
```
Comment (whoojemaflip) [inline spec/lib/nomis/client_spec.rb]: is this a likely response?

Comment (dan-palmer) [inline spec/lib/nomis/client_spec.rb]: I've seen the current sandbox return a servlet error as plain text error on a few occasions.

File: lib/nomis/client.rb (hunk 1 above)
Comment (dan-palmer) [inline lib/nomis/client.rb]: It defs should.

**Comment by whoojemaflip** [comment]
thumbs the hell up

**Pull request merged**